src/main/java/org/jboss/msc/service/ServiceControllerImpl.java

```diff
@@ -407,7 +413,9 @@ private Transition getTransition() {
         assert holdsLock(this);
         switch (state) {
             case NEW: {
-                return Transition.NEW_to_DOWN;
+                if (!getServiceContainer().isShutdown()) {
+                    return Transition.NEW_to_DOWN;
+                }
```

Comment:
This falls through; there should either be a ``break``, a ``throw``, a ``return``, or a ``/* fall through */`` comment.